go-indexer/indexer/indexer.go

```
@@ -172,202 +163,228 @@ func (i *Indexer) Start() error {
-		// Check if context is cancelled before processing each event
-		select {
-		case <-i.ctx.Done():
-			return fmt.Errorf("context cancelled during historical event processing")
-		default:
+	// listen for errors from subscription go routines
+	go func() {
+		for {
+			select {
+			case err := <-errors:
+				i.logger.Error("Error: %v", err)
+				i.Stop()
```

golangci-lint (1.64.8)

Error return value of `i.Stop` is not checked

(errcheck)